src/auth/roles.guard.ts

```diff
@@ -11,11 +11,22 @@ export class RolesGuard implements CanActivate {
     private readonly logger = new Logger(RolesGuard.name);
 
     canActivate(context: ExecutionContext): boolean {
-        const roleRequired = this.reflector.get<string>("roles", context.getHandler());
+        const roleRequiredMethod = this.reflector.get<string>(
+            RolesMetaData,
+            context.getHandler()
+        );
+        const roleRequiredClass = this.reflector.get<string>(
+            RolesMetaData,
+            context.getClass()
+        );
+        const roleRequired = roleRequiredMethod ?? roleRequiredClass;
+
         if (!roleRequired) {
             return true;
         }
+
         const request = context.switchToHttp().getRequest();
+        // TODO: This should either be JWT or Api key request.
```

Comment:
Remove todo. Each API key is tied to a hidden user with permissions. Thus, the backend doesn't have to differentiate between API key and client request.